roles/kubeadm_configs_compute/tasks/main.yaml

```diff
@@ -55,12 +55,41 @@
   changed_when: false
   check_mode: false
   delegate_to: '{{ groups.cp_running|first }}'
-  register: in_cluster_kubelet_config
+  register: in_cluster_kubelet_config_versionned
   when:
     - found_kubectl.rc == 0
     - groups.cp_running|default([])|length > 0
+    - current_cp_version is version("1.25", "<")
   environment:
     KUBECONFIG: /etc/kubernetes/admin.conf
+  ignore_errors: '{{ current_cp_version is version("1.23", ">=") }}'
+
+- name: 'Get unversionned kubelet configmap if cluster running'
+  command: >-
+    kubectl -n kube-system get cm
+    -o jsonpath='{.data.kubelet}'
+    kubelet-config
+  run_once: true
+  changed_when: false
+  check_mode: false
+  delegate_to: '{{ groups.cp_running|first }}'
+  register: in_cluster_kubelet_config_unversionned
+  when:
+    - found_kubectl.rc == 0
+    - groups.cp_running|default([])|length > 0
+    - current_cp_version is version("1.23", ">=")
+    - in_cluster_kubelet_config_versionned is skipped or in_cluster_kubelet_config_versionned.rc != 0
+  environment:
+    KUBECONFIG: /etc/kubernetes/admin.conf
+
+- name: 'Get kubelet configmap if cluster running'
```

Comment:
Are older configmap removed ?
What about getting  the unversionned configmap first (if cluster is in a verson where it could work), and falling back to versionned one if not found ?